src/aind_metadata_mapper/ephys/session.py

```diff
@@ -0,0 +1,147 @@
+        stage_logs : List
+          stage logs of all ephys data streams in a session
+        openephys_logs : List
+          openephys logs of all ephys data streams in a session
+        """
+        super().__init__(input_source, output_directory)
+        self.stage_logs = stage_logs
+        self.openephys_logs = openephys_logs
+        self.experiment_data = experiment_data
+
+    def _transform(self, extracted_source: ParsedInformation) -> Session:
+        """
+        Parses params from stage_log and openephys_log and
+        creates partial ephys session model
+        Parameters
+        ----------
+        extracted_source : ParsedInformation
+
+        Returns
+        -------
+        Session
+
+        """
+
+        stage_logs = extracted_source.stage_logs
+        openephys_logs = extracted_source.openephys_logs
+        experiment_data = extracted_source.experiment_data
+
+        ephys_session = {}
+
+        # Process data from dictionary keys
+        start_time = (
+            openephys_logs[0]
+            .getElementsByTagName("DATE")[0]
+            .firstChild.nodeValue
+        )
+        ephys_session["session_start_time"] = datetime.strptime(
+            start_time, "%d %b %Y %H:%M:%S"
+        )
+        ephys_session["experimenter_full_name"] = experiment_data[
+            "experimenter_full_name"
+        ]
+        ephys_session["session_end_time"] = experiment_data["session_end_time"]
+        ephys_session["subject_id"] = experiment_data["subject_id"]
+        ephys_session["session_type"] = experiment_data["session_type"]
+        ephys_session["iacuc_protocol"] = experiment_data["iacuc_protocol"]
+        ephys_session["rig_id"] = experiment_data["rig_id"]
+        ephys_session["animal_weight_prior"] = experiment_data[
+            "animal_weight_prior"
+        ]
+        ephys_session["maintenance"] = experiment_data["maintenance"]
+        ephys_session["calibrations"] = experiment_data["calibrations"]
+
+        # Constant throughout data streams
+        stick_microscopes = experiment_data["stick_microscopes"]
+        camera_names = experiment_data["camera_names"]
+        daqs = experiment_data["daqs"]
+        ephys_session["data_streams"] = []
+
+        for stage, data_stream in zip(
+            stage_logs, experiment_data["data_streams"]
+        ):
+            session_stream = {}
+            session_stream["stream_start_time"] = datetime.strptime(
+                stage[0][0], "%Y/%m/%d %H:%M:%S.%f"
+            )
+            session_stream["stream_end_time"] = datetime.strptime(
+                stage[-1][0], "%Y/%m/%d %H:%M:%S.%f"
+            )
+            session_stream["mouse_platform_name"] = data_stream[
+                "mouse_platform_name"
+            ]
+            session_stream["active_mouse_platform"] = data_stream[
+                "active_mouse_platform"
+            ]
+            session_stream["stream_modalities"] = data_stream[
```

Comment:
Can we assume that all streams are ephys streams? Or does the log also contain video/etc streams?

Reply:
I'm not sure I will ask Sasha